### PR TITLE
Added feature engineering .py file and logged accordingly

### PR DESCRIPTION
**Pull request by emchin**
Hi everyone! Here's my newest PR:

Here are the significant changes I made:
- Added a file called fe.py which adds a feature engineering step to the pipeline
- Changed nrows to 50,000 (instead of 1M)
- Added `df = fe(df)` as a step in the pipeline
- Added comments on different .py files

I'll clean up all other changes here. Please let me know your thoughts/feedback in the comments!

### CODE REVIEW

**Review by zaki1503 — APPROVED** [review]
Changes look good, especially the logger statements will be very helpful moving forward!

File: src/risknet/proc/reducer.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def reduce(fm_root, i):
                             'postal_code', 'mortgage_insurance_percent', 'original_loan_term']
 
     df = pd.concat([Reducer.simple_ts_split(pd.read_csv(fm_root + 'historical_data_2009Q1.txt', sep='|', index_col=False,
-                                                            names=origination_cols, nrows=1_000_000).merge(
+                                                            names=origination_cols, nrows=500_000).merge(
```
Comment (tambat22) [inline src/risknet/proc/reducer.py:34]: Im a little confused. Why did you change nrows to 500,000? Other than that, everything looks good!

Comment (emchin) [reply src/risknet/proc/reducer.py:34]: My feature engineering step added more dimensions (features) to the data. I reduced the number of rows because I was worried about using up too much memory!

I plan to test if n=1_000_000 is fine this week.

Comment (tambat22) [reply src/risknet/proc/reducer.py:34]: Ohh that makes sense. I tested n=1_000_000 on DSMLP and it works fine. But this is a minor issue. I'll approve the changes so far.

**Comment by tambat22** [comment]
Looks Good!

**Review by tambat22 — APPROVED** [review]
Looks Good!

**Review by Zhfang01 — APPROVED** [review]
great job!
